### PR TITLE
Fixes #1978. Add processing of HINT.UNREACHABLE_SWITCH_CASE

### LINKED ISSUE #1978: Update tests for analyzer reachability warnings
I'm [updating the test runner](https://dart-review.googlesource.com/c/sdk/+/291967) so that we can write static error tests of unreachable case warnings (well, hints currently) from analyzer. Before this change, there's no way to write a language or co19 test that validates that analyzer reports unreachable case warnings.

This is, unsurprisingly, breaking a few tests. I've updated the ones in the Dart repo, but there are some co19 ones too:

* co19/Language/Statements/Switch/execution_case_no_default_t01
* co19/Language/Statements/Switch/execution_case_t01
* co19/LanguageFeatures/Patterns/cast_A01_t02
* co19/LanguageFeatures/Patterns/constant_A02_t15
* co19/LanguageFeatures/Patterns/constant_A02_t18
* co19/LanguageFeatures/Patterns/logical_or_A05_t05
* co19/LanguageFeatures/Patterns/null_assert_A03_t01
* co19/LanguageFeatures/Patterns/null_assert_A03_t02
* co19/LanguageFeatures/Patterns/null_check_A02_t01
* co19/LanguageFeatures/Patterns/object_A03_t01
* co19/LanguageFeatures/Patterns/object_A07_t05
* co19/LanguageFeatures/Patterns/pattern_context_A07_t02
* co19/LanguageFeatures/Patterns/pattern_variable_sets_A01_t03
* co19/LanguageFeatures/Patterns/pattern_variable_sets_A02_t01
* co19/LanguageFeatures/Patterns/scope_A04_t01
* co19/LanguageFeatures/Patterns/scope_A04_t02
* co19/LanguageFeatures/Patterns/shared_case_scope_A01_t01
* co19/LanguageFeatures/Patterns/shared_case_scope_A01_t02
* co19/LanguageFeatures/Patterns/shared_case_scope_A01_t03
* co19/LanguageFeatures/Patterns/switch_statement_A01_t01
* co19/LanguageFeatures/Patterns/type_inference_A25_t01
* co19/LanguageFeatures/Patterns/type_inference_A31_t01
* co19/LanguageFeatures/Patterns/variable_A03_t02
* co19/LanguageFeatures/Patterns/variable_A03_t03
* co19/LanguageFeatures/Patterns/variable_A03_t05

The linked CL should give some guidance on how to fix them. It's probably some combination of:

* If it's already a static error test, then go ahead and add expectations for the unreachability errors too. They look like:

    ```dart
        case A(a: Enum.a, b: false): // Unreachable
    //  ^^^^
    // [analyzer] HINT.UNREACHABLE_SWITCH_CASE
    ``` 

* If the cases are intended to be unrelated and are just put into a single switch for convenience, it may be easiest to split them into separate switch statements to make the errors go away. That's what I did with invalid_const_pattern_test.dart.

Let me know if you have any questions or if there's anything I can do to help. As always, thanks. :)

### PR DESCRIPTION
**Pull request by sgrekhov**

### CODE REVIEW

File: LanguageFeatures/Patterns/switch_statement_A01_t01.dart [hunk 1]
```diff
@@ -30,32 +30,38 @@ String test(int value) {
...
-    case 30:
-      return "constant-2";
-    case (40):
-      return "parenthesized-1";
-    case (40):
-      return "parenthesized-2";
-    default:
-      return "default";
-  }
+  return "logical-and-1";
```
Comment (eernstg) [inline LanguageFeatures/Patterns/switch_statement_A01_t01.dart:33]: The indentation seems to have been lost from line 33 to 54.

File: LanguageFeatures/Patterns/switch_statement_A01_t01.dart [hunk 2]
```diff
@@ -30,32 +30,38 @@ String test(int value) {
...
+  default:
+  return "default";
+}
 }
 
 String testCast(num value) {
   switch (value) {
     case var c1 as double:
-      return "cast-1";
+    return "cast-1";
```
Comment (eernstg) [inline LanguageFeatures/Patterns/switch_statement_A01_t01.dart:60]: Indentation again.

File: LanguageFeatures/Patterns/switch_statement_A01_t01.dart [hunk 3]
```diff
@@ -30,32 +30,38 @@ String test(int value) {
...
     case var c2 as double:
-      return "cast-2";
+//  ^^^^
+// [analyzer] HINT.UNREACHABLE_SWITCH_CASE
+    return "cast-2";
```
Comment (eernstg) [inline LanguageFeatures/Patterns/switch_statement_A01_t01.dart:64]: Indentation.

File: LanguageFeatures/Patterns/switch_statement_A01_t01.dart [hunk 4]
```diff
@@ -64,34 +70,40 @@ String testCast(num value) {
...
   switch (value) {
     case var a1?:
-      return "null-check-1";
-    case var a2?:
-      return "null-check-2";
-    default:
-      return "default";
-  }
+  return "null-check-1";
+  case var a2?:
```
Comment (eernstg) [inline LanguageFeatures/Patterns/switch_statement_A01_t01.dart:74]: Somehow the indentation has been messed up in many places! ;-)

**Review by eernstg — COMMENTED** [review]
Something happened to the indentation, I think it starts in LanguageFeatures/Patterns/switch_statement_A01_t01.dart. Otherwise it looks good!

**Comment by sgrekhov** [comment]
Fixed. Please take another look

**Review by eernstg — APPROVED** [review]
LGTM

**Review by munificent — COMMENTED** [review]
LGTM, thanks!

**Pull request merged**